src/mscorlib/src/System/IO/IsolatedStorage/IsolatedStorageFileSmall.cs

```diff
@@ -1772,7 +1772,7 @@ internal void Lock(ref bool locked)
                 // Calls to GetLastWin32Error overwrites HResult.  Store HResult.
                 hr = Marshal.GetLastWin32Error();
                 if(hr == Win32Native.ERROR_FILE_NOT_FOUND)
-                    return new String[0];
+                    return Array.Empty<String>();
```

Comment:
IsolatedStorage isn't currently exposed in .NET Core, and even if it were, this is an error path we don't need to optimize.  This is also returning the array out of a public API, which means there's non-zero risk (though I expect it's very minimal). Given that, we should revert this.